### PR TITLE
Prepare for dated newspaper page requests

### PR DESCRIPTION
**Pull request by JustinPinner**
This PR adds handling of newspaper book and book section requests so that we no longer have to hand these off to R2.

Tests are misbehaving locally, so I'm using the PR to trigger a server-side build and test :sunglasses:

### CODE REVIEW

File: applications/app/views/newspaperPages.scala.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+@(index: services.IndexPage, nav: PreviousAndNext)(implicit request: RequestHeader)
```
Comment (jennysivapalan) [inline applications/app/views/newspaperPages.scala.html]: This view isn't used so can be removed?

Comment (JustinPinner) [inline applications/app/views/newspaperPages.scala.html]: Ah yes, thanks. That was left over from the earliest days of this thing's existence. Should have been binned a while back. Will do that now.

File: article/app/controllers/PublicationController.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        s"$publication/${urlFormat(reqDate)}/$tail"
+      } else {
+        s"$publication/${urlFormat(reqDate)}"
+      }
+      ArticleController.renderItem(newPath)
+    }
+
+  }
+
+  private def tagExists(publication: String, tag: String) = {
```
Comment (jennysivapalan) [inline article/app/controllers/PublicationController.scala]: might be worth renaming to `bookSectionTagExists`

File: article/conf/routes [hunk 3]
```diff
@@ -9,6 +9,17 @@ GET     /_healthcheck               conf.HealthCheck.healthcheck()
 
 GET     /_cdn_healthcheck           controllers.CdnHealthcheckController.healthcheck()
 
+# Newspaper pages paths
+# gallery format (?)
+#  e.g. /theobserver/gallery/2013/sep/14/the-10-best-fonts
```
Comment (jennysivapalan) [inline article/conf/routes]: Is gallery format catered for?

File: article/test/PublicationControllerTest.scala [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+//  it should "redirect to an /all page when an observer dated blog section is requested" in {
+//    when(bookAgent.getTags("theobserver/she-said")).thenReturn(emptySeq)
+//    when(bookSectionAgent.getTags("theobserver/she-said")).thenReturn(emptySeq)
+//    val testReq = TestRequest("theobserver/she-said/2015/jun/15")
+//    val result = controllers.PublicationController.publishedOn("theobserver/she-said","2015","jun","15","")(testReq)
+//    status(result) should be(PermanentRedirect)
+//    header("Location",result).getOrElse("") should endWith ("/theobserver/she-said/2015/jun/15/all")
+//  }
```
Comment (jennysivapalan) [inline article/test/PublicationControllerTest.scala]: Remove this one?

File: article/test/PublicationControllerTest.scala [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+//  it should "redirect to an /all page when a guardian dated blog section is requested" in {
+//    when(bookAgent.getTags("theguardian")).thenReturn(emptySeq)
+//    when(bookSectionAgent.getTags("theguardian")).thenReturn(emptySeq)
+//    val testReq = TestRequest("theguardian/from-the-archive-blog/2012/feb/17")
+//    val result = controllers.PublicationController.publishedOn("theguardian/from-the-archive-blog","2012","feb","17","")(testReq)
+//    status(result) should be(PermanentRedirect)
+//    header("Location",result).getOrElse("") should endWith ("/theguardian/from-the-archive-blog/2012/feb/17/all")
+//  }
+
```
Comment (jennysivapalan) [inline article/test/PublicationControllerTest.scala]: Remove this one?

File: data/database/b083e89694f062bff53fc6c3e0f1e4a8027a77f07c8e18e7d4b9d72e4c281ab9 [hunk 6]
```diff
@@ -0,0 +1 @@
+Error:404
```
Comment (jennysivapalan) [inline data/database/b083e89694f062bff53fc6c3e0f1e4a8027a77f07c8e18e7d4b9d72e4c281ab9]: None of the tests you created generate a 404 so are these needed?

**Comment by jennysivapalan** [comment]
:+1:

File: article/app/controllers/PublicationController.scala (hunk 2 above)
Comment (JustinPinner) [inline article/app/controllers/PublicationController.scala]: Fair point. Will do.

File: article/conf/routes (hunk 3 above)
Comment (JustinPinner) [inline article/conf/routes]: The gallery url works so if CP are happy we're ok.

File: article/test/PublicationControllerTest.scala (hunk 4 above)
Comment (JustinPinner) [inline article/test/PublicationControllerTest.scala]: This is one of those that won't test properly. They work when running (locally) so I'd like to make sure I don't forget this pattern when testing on code. Maybe I'll pop a reminder on the Trello card and ditch these from the test for now.

File: article/test/PublicationControllerTest.scala (hunk 5 above)
Comment (JustinPinner) [inline article/test/PublicationControllerTest.scala]: As above - will remove and make a note in Trello to test manually.

File: data/database/b083e89694f062bff53fc6c3e0f1e4a8027a77f07c8e18e7d4b9d72e4c281ab9 (hunk 6 above)
Comment (JustinPinner) [inline data/database/b083e89694f062bff53fc6c3e0f1e4a8027a77f07c8e18e7d4b9d72e4c281ab9]: The commented tests produce the 404s :cry: That's how these got here. I'll ditch them though.

**Pull request merged**